tests/lib/cli-engine.js

```diff
@@ -1812,6 +1813,74 @@ describe("CLIEngine", function() {
                 assert.deepEqual(result, cachedResult, "result is the same with or without cache");
             });
 
+            it("should not delete cache when executing on text", function() {
```

Comment:
Is it worth also including a test for `should not delete cache when executing on text with a provided filename`, to address https://github.com/eslint/eslint/issues/6459#issuecomment-228552473 ?